src/backend/ctrl_flow_struct/condition.rs

```diff
@@ -0,0 +1,171 @@
+use std::fmt;
```

Comment:
Please add a high level description of this module in doc comments so that it shows up in rustdocs.

Things to include:
- What does this module provide, where does it fit into the control flow restructuring work.
- High-level API design and how this module is designed to be used.